File: frontend/src/app/core/services/ranking.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CategoriaJugador } from '../../shared/models/torneo.model';
import { RankingJugadorResponse } from '../../shared/models/ranking.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class RankingService {
  private http = inject(HttpClient);
  private readonly API_URL = environment.apiUrl;

  obtenerRankingPorCategoria(categoria: CategoriaJugador): Observable<RankingJugadorResponse[]> {
    return this.http.get<RankingJugadorResponse[]>(`${this.API_URL}/rankings/${categoria}`);
  }

  obtenerMiPosicion(categoria: CategoriaJugador): Observable<RankingJugadorResponse> {
    return this.http.get<RankingJugadorResponse>(`${this.API_URL}/rankings/${categoria}/mi-posicion`);
  }
}

File: backend/src/main/java/com/padel/service/RankingService.java
package com.padel.service;

import com.padel.domain.enums.CategoriaJugador;
import com.padel.dto.response.RankingJugadorResponse;

import java.util.List;

public interface RankingService {
    List<RankingJugadorResponse> obtenerRankingPorCategoria(CategoriaJugador categoria);
    RankingJugadorResponse obtenerMiPosicion(CategoriaJugador categoria, String usuarioEmail);
}
